### PR TITLE
mustache-tcl: init at 1.1.3.4

### PR DESCRIPTION
**Pull request by nat-418**
###### Description of changes

Add Tcl package / library for handling mustache templates.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/mustache-tcl/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ lib, fetchzip, tcl, tcllib }:
+
+tcl.mkTclDerivation {
+  pname = "mustache-tcl";
+  version = "1.1.3.4";
+
+  src = fetchzip {
+    url = "https://github.com/ianka/mustache.tcl/archive/refs/tags/v1.1.3.4.zip";
```
Comment (JulienMalka) [inline pkgs/development/libraries/mustache-tcl/default.nix:8]:
```suggestion
    url = "https://github.com/ianka/mustache.tcl/archive/refs/tags/v${version}.zip";
```

**Review by JulienMalka — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/mustache-tcl/default.nix (hunk 1 above)
Comment (JulienMalka) [reply pkgs/development/libraries/mustache-tcl/default.nix:8]: By the way why are you not using fetchFromGitHub ?

Comment (nat-418) [reply pkgs/development/libraries/mustache-tcl/default.nix]: No reason, I will change to that if you want.

**Comment by JulienMalka** [comment]
Result of `nixpkgs-review pr 231351` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mustache-tcl</li>
  </ul>
</details>

**Review by JulienMalka — APPROVED** [review]

File: pkgs/development/libraries/mustache-tcl/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    mkdir -p $out/lib/mustache-tcl
+    cp $src/mustache.tcl $out/lib/mustache-tcl/mustache.tcl
+    cp $src/pkgIndex.tcl $out/lib/mustache-tcl/pkgIndex.tcl
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/ianka/mustache.tcl";
+    description = "Tcl implementation of the mustache templating language";
+    license = licenses.bsd2;
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/mustache-tcl/default.nix:28]:
```suggestion
    platforms = tcllib.meta.platforms;
```

**Comment by SuperSandro2000** [comment]
@ofborg build mustache-tcl

**Pull request merged**